tests/e2e/utils/Logger.ts

```diff
@@ -10,82 +10,113 @@
-     * @param text log text
-     * @param indentLevel log level
-     */
-    static debug(text: string, indentLevel: number = 5): void {
-        if (ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'ERROR' ||
-            ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'WARN' ||
-            ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'INFO') {
-            return;
-        }
-        this.logText(indentLevel, `▼ ${text}`);
+  /**
+   * Uses for logging of the public methods of the {@link DriverHelper} or
+   * private methods inside of pageobjects.
+   * @param text log text
+   * @param indentLevel log level
+   */
+  static trace(text: string = '', indentLevel: number = 6): void {
+    if (
+      ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'ERROR' ||
+      ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'WARN' ||
+      ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'INFO' ||
+      ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'DEBUG'
+    ) {
+      return;
     }
+    this.logText(
+      indentLevel,
+      `‣ ${this.getCallerInfo()} ${this.separator(text)} ${text}`
+    );
+  }
 
-    /**
-     * Uses for logging of the public methods of the {@link DriverHelper} or
-     * private methods inside of pageobjects.
-     * @param text log text
-     * @param indentLevel log level
-     */
-    static trace(text: string, indentLevel: number = 6): void {
-        if (ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'ERROR' ||
-            ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'WARN' ||
-            ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'INFO' ||
-            ReporterConstants.TS_SELENIUM_LOG_LEVEL === 'DEBUG') {
-            return;
-        }
-        this.logText(indentLevel, `‣ ${text}`);
+  private static logText(messageIndentationLevel: number, text: string): void {
+    // start group for every level
+    for (let i: number = 0; i < messageIndentationLevel; i++) {
+      console.group();
     }
-
-    private static logText(messageIndentationLevel: number, text: string): void {
-        // start group for every level
-        for (let i: number = 0; i < messageIndentationLevel; i++) {
-            console.group();
-        }
-        // print the trimmed text
-        // if multiline, the message should be properly padded
-        console.log(text);
-        // end group for every level
-        for (let i: number = 0; i < messageIndentationLevel; i++) {
-            console.groupEnd();
-        }
+    // print the trimmed text
+    // if multiline, the message should be properly padded
+    console.log(text);
+    // end group for every level
+    for (let i: number = 0; i < messageIndentationLevel; i++) {
+      console.groupEnd();
     }
+  }
+
+  private static getCallerInfo(): string {
+    const e: Error = new Error();
+    const stack: string[] = e.stack ? e.stack.split('\n') : [];
+    // " at functionName ( ..." => "functionName"
+    return stack[3].replace(/^\s+at\s+(.+?)\s.+/g, '$1');
```

Comment:
What about don't logging `Context.<anonymous>` and `Runner.<anonymous>` to make log cleaner? 
```
stack[3].replace(/^\s+at\s+(.+?)\s.+/g, '$1')
            .replace('Context.<anonymous>', '')
            .replace('Runner.<anonymous> ', '')
```

E.g.:
```
    ✔ Get recommended extensions list from extensions.json (10623ms)
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
     ✔ Open "Extensions" view section (1885ms)
       • Context.<anonymous> - Time for extensions installation TimeoutConstants.TS_COMMON_PLUGIN_TEST_TIMEOUT=30000
             ‣ DriverHelper.wait - (30000 milliseconds)
     ✔ Let extensions complete installation (30003ms)
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
     1) Check if extensions are installed and enabled
   [ERROR] Runner.<anonymous> - CheReporter runner.on.fail: Check if recommended extensions installed for Ansible Check if extensions are installed and enabled failed after 3346ms
             ‣ Runner.<anonymous> - FullTitle:Check if recommended extensions installed for Ansible Check if extensions are installed and enabled
             ‣ Runner.<anonymous> - FullTitleSanitized:Check_if_recommended_extensions_installed_for_Ansible_Check_if_extensions_are_installed_and_enabled
             ‣ Runner.<anonymous> - TestTitle:Check if extensions are installed and enabled
             ‣ Runner.<anonymous> - TestTitleSanitized:Check_if_extensions_are_installed_and_enabled
             ‣ DriverHelper.getDriver  
```

Reply:
https://github.com/eclipse/che/issues/22448